### ISSUE TITLE
[SUGGESTION] Avoid platform specific code in regression tests

### ISSUE THREAD

**Issue by filipsajdak** [hsutter/cppfront#166]
I am happy to see execution output as well as output from the compiler in the `regression-tests/test-results/{clang-12,gcc-10,msvc-2022}` directories. 

Current tests used expressions that create highly platform-dependent results that will be hard to test against. 

E.g.:
```cpp
typeid(x).name()
```
will generate various mangled names depending on the platform. 

As an example after running tests on my macOS with clang-14 I have:
```diff
diff --git a/regression-tests/test-results/clang-12/pure2-type-safety-1.cpp.execution b/regression-tests/test-results/clang-12/pure2-type-safety-1.cpp.execution
index 6e21ead..7713008 100644
--- a/regression-tests/test-results/clang-12/pure2-type-safety-1.cpp.execution
+++ b/regression-tests/test-results/clang-12/pure2-type-safety-1.cpp.execution
@@ -1,9 +1,9 @@
                               d is int? false
-               St7variantIJidEE is int? false
-                         St3any is int? false
-                 St8optionalIiE is int? false
+         NSt3__17variantIJidEEE is int? false
+                   NSt3__13anyE is int? false
+           NSt3__18optionalIiEE is int? false
 
                               i is int? true
-               St7variantIJidEE is int? true
-                         St3any is int? true
-                 St8optionalIiE is int? true
+         NSt3__17variantIJidEEE is int? true
+                   NSt3__13anyE is int? true
+           NSt3__18optionalIiEE is int? true
```

That creates false positive test cases.

**Comment by hsutter** [comment]
Thanks! This should now be fixed by commit ec32963f1144174e39d4274ab72801cbc238c666 .

**Comment by filipsajdak** [comment]
Thanks! It will make my testing easier.
